packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts

```diff
@@ -143,7 +143,7 @@ function setupTestSuiteMatrix(
           }
         }
         clients.length = 0
-        if (options?.skipDb !== true || (options?.skipDb !== true && process.env.JEST_MAX_WORKERS !== '1')) {
+        if (options?.skipDb !== true && process.env.TEST_REUSE_DATABASE !== 'true' && process.env.CI !== 'true') {
```

Comment:
Only drops the db if no skipped, and if the db does not need to be reused. And also, if we are on CI why care to waste time dropping the db, also avoid dropping it.